### PR TITLE
Added:Google PlaceAutocomplete

### LINKED ISSUE #799: Add Google Place AutoComplete
**Describe the feature you'd like**
For searching events in a specific place, we should use google place autocomplete API.

**Would you like to work on the issue?**
- [X] Yes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #799 

Changes: Added Google PlaceAutocomplete for searching events 
Screenshots for the change:

videotogif_2019 01 02_01 15 01

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal please review

File: app/build.gradle [hunk 1]
```diff
@@ -102,6 +106,8 @@ dependencies {
     testImplementation "androidx.room:room-testing:${roomVersion}"
     implementation "com.takisoft.fix:preference-v7:28.0.0.0"
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'com.google.android.gms:play-services-places:16.0.0'
```
Comment (iamareebjamal) [inline app/build.gradle]: Read comments from this PR - https://github.com/fossasia/open-event-android/pull/735

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal i need to disable the search functionality for F-Droid application right?
but when i tried it with playStoreImplementation i'm not able to build the project as class not found for Places API shows up

**Comment by aggarwalpulkit596** [comment]
@nikit19 i will handle the case for f- droid but if i use playStoreImplementation i'm not able to import the classes required? what's the solution to that

**Comment by nikit19** [comment]
What do you mean by handle the case for Fdroid? You can use those imports only in the playstore flavor. Please read about build flavors, you will get a better understanding of how to handle this case.

**Comment by aggarwalpulkit596** [comment]
@nikit19 i meant that if the build type if of play store then only I'll do this process otherwise we can do the previous method

**Comment by nikit19** [comment]
Yes, but you are not doing that in the code. Try building the project for the playstore flavor, the build will complete. Right now you are implementing the places API only in the playstore flavor but importing it in both the flavors.

**Comment by aggarwalpulkit596** [comment]
@nikit19 @iamareebjamal this is correct method i guess ?

File: app/build.gradle [hunk 2]
```diff
@@ -56,6 +56,13 @@ android {
         }
     }
 
+    variantFilter { variant ->
+        def names = variant.flavors*.name
+        if(variant.buildType.name == 'release' && names.contains("fdroid")) {
+            setIgnore(true)
+        }
+    }
```
Comment (iamareebjamal) [inline app/build.gradle]: Uneeded

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal can you look at the Travis error i don't find any reason for it to fail

**Comment by iamareebjamal** [comment]
The fix is literally written below the error https://travis-ci.com/fossasia/open-event-android/builds/96083368

**Comment by iamareebjamal** [comment]
```
* What went wrong:
Execution failed for task ':app:spotlessKotlin'.
> The following files had format violations:
      app/src/playStore/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt
          @@ -28,8 +28,8 @@
           ········val·bundle·=·intent.extras
           ········val·fromSearchFragment·=·bundle?.getBoolean(FROM_SEARCH)·?:·false
           
          -········val·autocompleteFragment·=·fragmentManager.
          -············findFragmentById(R.id.placeAutocompleteFragment)·as·PlaceAutocompleteFragment
          +········val·autocompleteFragment·=·fragmentManager
          +············.findFragmentById(R.id.placeAutocompleteFragment)·as·PlaceAutocompleteFragment
           
           ········autocompleteFragment.setOnPlaceSelectedListener(object·:·PlaceSelectionListener·{
           ············override·fun·onPlaceSelected(place:·Place)·{
          @@ -49,7 +49,6 @@
           
           ············override·fun·onError(status:·Status)·{
           ················Timber.d(status.statusMessage,·"Failed·getting·location")
          -
           ············}
           ········})
           ····}
  Run 'gradlew spotlessApply' to fix these violations.
```

File: app/build.gradle [hunk 3]
```diff
@@ -8,6 +8,8 @@ apply plugin: 'kotlin-kapt'
 apply plugin: 'kotlin-android-extensions'
 
 def STRIPE_API_TOKEN = System.getenv('STRIPE_API_TOKEN') ?: "YOUR_API_KEY"
+def GOOGLE_API_KEY = System.getenv('GOOGLE_API_KEY ') ?: "YOUR_API_KEY"
+
```
Comment (nikit19) [inline app/build.gradle]: extra

File: app/build.gradle [hunk 4]
```diff
@@ -21,7 +23,9 @@ android {
         vectorDrawables.useSupportLibrary = true
         multiDexEnabled true
         manifestPlaceholders = [
-                STRIPE_API_TOKEN: STRIPE_API_TOKEN
+                STRIPE_API_TOKEN: STRIPE_API_TOKEN,
+                GOOGLE_API_KEY: GOOGLE_API_KEY
+
```
Comment (nikit19) [inline app/build.gradle]: extra

File: app/build.gradle [hunk 5]
```diff
@@ -102,6 +106,8 @@ dependencies {
     testImplementation "androidx.room:room-testing:${roomVersion}"
     implementation "com.takisoft.fix:preference-v7:28.0.0.0"
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    playStoreImplementation 'com.google.android.gms:play-services-places:16.0.0'
+
```
Comment (nikit19) [inline app/build.gradle]: extra

File: app/src/fdroid/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt [hunk 6]
```diff
@@ -24,7 +24,6 @@ class SearchLocationActivity : AppCompatActivity() {
         this.supportActionBar?.title = ""
         val bundle = intent.extras
         val fromSearchFragment = bundle?.getBoolean(FROM_SEARCH) ?: false
-
```
Comment (nikit19) [inline app/src/fdroid/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt]: not required

File: app/src/playStore/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        val bundle = intent.extras
+        val fromSearchFragment = bundle?.getBoolean(FROM_SEARCH) ?: false
+
+        val autocompleteFragment = fragmentManager
+            .findFragmentById(R.id.placeAutocompleteFragment) as PlaceAutocompleteFragment
+
+        autocompleteFragment.setOnPlaceSelectedListener(object : PlaceSelectionListener {
+            override fun onPlaceSelected(place: Place) {
+                // Save Searched Place String
+                searchLocationViewModel.saveSearch(place.name as String)
```
Comment (nikit19) [inline app/src/playStore/java/org/fossasia/openevent/general/search/SearchLocationActivity.kt]: `.toString()` would work?

**Comment by aggarwalpulkit596** [comment]
@nikit19 i made two different files for f-droid and play store but github is only showing one file but it is actually two seperate files

**Pull request closed without merging**